### PR TITLE
[ API ] Train API with data files

### PR DESCRIPTION
**Pull request by jijoongmoon**
Now two Options for the Train are available.
- ml_nnmodel_train_with_file(ml_nnmodel_h)
  : Suppose every parameters are set from configuraiton and data
  files ( Training : mandatory , Validataion : optional, Test :
  optional )
- ml_nnmodel_train_with_file(ml_nnmodel_h, ...)
  : Suppose using APIs for the set up the network configuration and
    Network hyper parameters are given by arguments (key = value
    format )

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/src/databuffer_file.cpp [hunk 1]
```diff
@@ -250,6 +249,7 @@ int DataBufferFromDataFile::setDataFile(std::string path, DataType type) {
     train_name = path;
   } break;
```
Comment (again4you) [inline nntrainer/src/databuffer_file.cpp:250]: IMO, there is a bug. Need to add `validation[type] = false;` and remove `train_name = path;` code from `if` block

File: nntrainer/src/fc_layer.cpp [hunk 2]
```diff
@@ -188,7 +188,7 @@ int FullyConnectedLayer::setProperty(std::vector<std::string> values) {
       weight_ini_type = (WeightIniType)parseType(value, TOKEN_WEIGHTINI);
       break;
     default:
-      ml_loge("Error: Unknown Layer Property Key");
+      ml_loge("Error: Unknown Layer Property Key : %s", key);
```
Comment (again4you) [inline nntrainer/src/fc_layer.cpp:191]: IMO, `key.c_str()`is much better.

File: nntrainer/src/databuffer_file.cpp (hunk 1 above)
Comment (again4you) [reply nntrainer/src/databuffer_file.cpp:250]: My mistake. LGTM!

File: nntrainer/src/neuralnet.cpp [hunk 3]
```diff
@@ -364,30 +365,75 @@ int NeuralNetwork::init() {
...
     case PropertyType::loss: {
       cost = (CostType)parseType(value, TOKEN_COST);
     } break;
+    case PropertyType::batch_size: {
+      status = setInt(batch_size, value);
+      NN_RETURN_STATUS();
+      for (unsigned int i = 0; i < layers.size(); ++i) {
+        if (layers[i]->getTensorDim().batch() !=
+            static_cast<unsigned int>(batch_size)) {
+          ml_logw("Worning: Batch Size is changing!! : %d -> %d",
```
Comment (zhoonit) [inline nntrainer/src/neuralnet.cpp:394]:
```suggestion
          ml_logw("Warning: Batch Size is changing!! : %d -> %d",
```

**Pull request merged**